elastic/eui
# [Emotion] Convert EuiButtonEmpty

## Summary

This PR:

- Converts `EuiButtonEmpty` to Emotion, removing/cleaning up unnecessary CSS due to dogfooding of new internal `EuiButtonDisplay` components
- Removes all `.euiButtonEmpty--` modifier classes
- Removes `EuiButtonContentDeprecated` fully - this was the last component using the old/deprecated internal button components
- closes https://github.com/elastic/eui/issues/6388 - this is the last button component to be converted to Emotion 🎉 

As always, for review I recommend [following along by commit](https://github.com/elastic/eui/pull/6863/commits) (due to the number of affected snapshots).

## QA

- Confirm that [production](https://elastic.github.io/eui/#/navigation/button#empty-button) looks similar to [staging](https://eui.elastic.co/pr_6863/#/navigation/button#empty-button) (note that there is an expected slight font/typography change)
- `gh pr checkout 6863 && yarn storybook`
- Confirm that the `color`, `flush`, and `size` permutations work as expected in Storybook

### General checklist


## Emotion checklist

### Acceptance criteria
~- [ ] All SCSS overrides have been removed from the Amsterdam theme~ - No overrides for this component
~- [ ] Any dependent components are identified in a new issue~ - All dependent usages in EUI are calling the generic `.euiButtonEmpty` or `.euiButtonEmpty__content`

### Checklists

**Kibana usage**

**Kibana due diligence**

~- [ ] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted~ None
  ~- [ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~ - Usages can be swapped out for non-modifier `.euiButtonEmpty`
    - Multiple updates needed
    - [Two](https://github.com/elastic/kibana/blob/c339ca839405eb708ad628e2ce2b5f2f1747fdb6/x-pack/plugins/cases/public/components/configure_cases/connectors_dropdown.tsx#L51-L55) [usages](https://github.com/elastic/kibana/blob/c339ca839405eb708ad628e2ce2b5f2f1747fdb6/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_empty_prompt.tsx#L95-L101) that will need to be fixed

---

**General**


---

**Unit tests**

- [x] Converted from Enzyme to RTL
~- [ ] Removed any mounted snapshots~

---

**Sass/Emotion conversion process**


~- [ ] Deleted any `src/amsterdam/overrides/{component}.scss` files (styles within should have been converted to the baseline Emotion styles)~
~- [ ] Converted all global Sass vars/mixins to JS (e.g. `$euiSize` to `euiTheme.size.base`)~
~- [ ] Removed or converted component-specific Sass vars/mixins to exported JS versions~
~- [ ] Listed var/mixin removals in changelog~
~- [ ] Ran `yarn compile-scss` to update var/mixin [JSON files](https://github.com/elastic/eui/tree/main/src-docs/src/views/theme/_json)~
~- [ ] Simplified `calc()` to `mathWithUnits` if possible (if mixing different unit types, this may not be possible)~
~- [ ] Added an `@warn` deprecation message within the `global_styling/mixins/{component}.scss` file~

---

**CSS tech debt**

~- [ ] Used `gap` property to add margin between items if using flex~

---

**DOM Cleanup**


---

**Extras/nice-to-have**

~- [ ] Documentation pass~
~- [ ] Check for issues in the backlog that could be a quick fix for that component~
~- [ ] Optional component/code cleanup: consider splitting up the component into multiple children if it's overly verbose or difficult to reason about~
